# reference path

#lisaru track1
track_points_x: [0.0, 0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1.0, 1.1, 1.2, 1.3, 1.4, 1.5, 1.6, 1.7, 1.8, 1.9, 2.0, 2.1, 2.2, 2.3, 2.4, 2.5, 2.6, 2.7, 2.8, 2.9, 3.0, 2.9, 2.8, 2.7, 2.6, 2.5, 2.4, 2.3, 2.2, 2.1, 2.0, 1.9, 1.8, 1.7, 1.6, 1.5, 1.4, 1.3, 1.2, 1.1, 1.0, 0.9, 0.8, 0.7, 0.6, 0.5, 0.4, 0.3, 0.2, 0.1, 0.0, -0.1, -0.2, -0.3, -0.4, -0.5, -0.6, -0.7, -0.8, -0.9, -1.0, -1.1, -1.2, -1.3, -1.4, -1.5, -1.6, -1.7, -1.8, -1.9, -2.0, -2.1, -2.2, -2.3, -2.4, -2.5, -2.6, -2.7, -2.8, -2.9, -3.0, -2.9, -2.8, -2.7, -2.6, -2.5, -2.4, -2.3, -2.2, -2.1, -2.0, -1.9, -1.8, -1.7, -1.6, -1.5, -1.4, -1.3, -1.2, -1.1, -1.0, -0.9, -0.8, -0.7, -0.6, -0.5, -0.4, -0.3, -0.2, -0.1, 0.0]
track_points_y: [0.0, 0.10, 0.20, 0.29, 0.39, 0.47, 0.56, 0.63, 0.70, 0.77, 0.83, 0.88, 0.93, 0.96, 1.00, 1.02, 1.04, 1.05, 1.06, 1.06, 1.05, 1.04, 1.01, 0.98, 0.93, 0.88, 0.81, 0.72, 0.60, 0.44, 0.0, -0.44, -0.60, -0.72, -0.81, -0.88, -0.93, -0.98, -1.01, -1.04, -1.05, -1.06, -1.06, -1.05, -1.04, -1.02, -1.00, -0.96, -0.93, -0.88, -0.83, -0.77, -0.70, -0.63, -0.56, -0.47, -0.39, -0.29, -0.20, -0.10, 0.0, 0.10, 0.20, 0.29, 0.39, 0.47, 0.56, 0.63, 0.70, 0.77, 0.83, 0.88, 0.93, 0.96, 1.00, 1.02, 1.04, 1.05, 1.06, 1.06, 1.05, 1.04, 1.01, 0.98, 0.93, 0.88, 0.81, 0.72, 0.60, 0.44, 0.0, -0.44, -0.60, -0.72, -0.81, -0.88, -0.93, -0.98, -1.01, -1.04, -1.05, -1.06, -1.06, -1.05, -1.04, -1.02, -1.00, -0.96, -0.93, -0.88, -0.83, -0.77, -0.70, -0.63, -0.56, -0.47, -0.39, -0.29, -0.20, -0.10, 0.0]



#lisaru track2
# track_points_x: [0.0, 0.2, 0.4, 0.6, 0.8, 1.0, 1.2, 1.4, 1.6, 1.8, 2.0, 2.2, 2.4, 2.6, 2.8, 3.0, 3.2, 3.4, 3.6, 3.8, 4.0, 4.2, 4.4, 4.6, 4.8, 5.0, 5.2, 5.4, 5.6, 5.8, 6.0, 5.8, 5.6, 5.4, 5.2, 5.0, 4.8, 4.6, 4.4, 4.2, 4.0, 3.8, 3.6, 3.4, 3.2, 3.0, 2.8, 2.6, 2.4, 2.2, 2.0, 1.8, 1.6, 1.4, 1.2, 1.0, 0.8, 0.6, 0.4, 0.2, 0.0, -0.2, -0.4, -0.6, -0.8, -1.0, -1.2, -1.4, -1.6, -1.8, -2.0, -2.2, -2.4, -2.6, -2.8, -3.0, -3.2, -3.4, -3.6, -3.8, -4.0, -4.2, -4.4, -4.6, -4.8, -5.0, -5.2, -5.4, -5.6, -5.8, -6.0, -5.8, -5.6, -5.4, -5.2, -5.0, -4.8, -4.6, -4.4, -4.2, -4.0, -3.8, -3.6, -3.4, -3.2, -3.0, -2.8, -2.6, -2.4, -2.2, -2.0, -1.8, -1.6, -1.4, -1.2, -1.0, -0.8, -0.6, -0.4, -0.2, 0.0]

# track_points_y: [0.0, 0.2, 0.4, 0.58, 0.78, 0.94, 1.12, 1.26, 1.4, 1.54, 1.66, 1.76, 1.86, 1.92, 2.0, 2.04, 2.08, 2.1, 2.12, 2.12, 2.1, 2.08, 2.02, 1.96, 1.86, 1.76, 1.62, 1.44, 1.2, 0.88, 0.0, -0.88, -1.2, -1.44, -1.62, -1.76, -1.86, -1.96, -2.02, -2.08, -2.1, -2.12, -2.12, -2.1, -2.08, -2.04, -2.0, -1.92, -1.86, -1.76, -1.66, -1.54, -1.4, -1.26, -1.12, -0.94, -0.78, -0.58, -0.4, -0.2, 0.0, 0.2, 0.4, 0.58, 0.78, 0.94, 1.12, 1.26, 1.4, 1.54, 1.66, 1.76, 1.86, 1.92, 2.0, 2.04, 2.08, 2.1, 2.12, 2.12, 2.1, 2.08, 2.02, 1.96, 1.86, 1.76, 1.62, 1.44, 1.2, 0.88, 0.0, -0.88, -1.2, -1.44, -1.62, -1.76, -1.86, -1.96, -2.02, -2.08, -2.1, -2.12, -2.12, -2.1, -2.08, -2.04, -2.0, -1.92, -1.86, -1.76, -1.66, -1.54, -1.4, -1.26, -1.12, -0.94, -0.78, -0.58, -0.4, -0.2, 0.0]


#track 2
# track_points_x: [0,-0.5,-2,-3,-1,1,2,1,0,-2,-4,-7,-11,-14,-16,-20]
# track_points_y: [0,2,4,6,8,10,12,14,16,15,14,16,18,17,16,16]

#track 3
# track_points_x: [0,0,-1,-2,-3,-4,-5,-10,]
# track_points_y: [5,10,13,14,14.583,14.889,15,15]


#track 4
# track_points_x: [-1,0,2,4,6,4,2,0,-2,-4,-6,-4,-2,-1]
# track_points_y: [0,0,0.34,1.53,6,10.47,11.66,12,11.66,10.47,6,1.53,0.34,-1]


# track real
# track_points_x: [0,1,2,1,0,1,2]
# track_points_y: [0,2,4,6,8,10,12]


desired_v: 0.8 # desired veloc1.0ity

dt: 0.05
N: 20 # horizon
Kp: 100.0
Ki: 0.0
Kd: 0.0

# # lower and upper bound
# v_max: 4
# a_max: 2
# delta_max: 2
# ddelta_max: 2

# ctrl delay
delay: 0.125

# true for nmpc, false for qp
use_serial: true
port: "/dev/feedback_controller"
baudrate: 115200
